src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsVocabularyAnnotation.cs

```diff
@@ -396,9 +396,9 @@ private IEdmVocabularyAnnotatable ComputeTarget()
                         // $ReturnType
                         if (targetSegments[1] == CsdlConstants.OperationReturnExternalTarget)
                         {
-                            if (operation.ReturnType != null)
+                            if (operation.Return?.Type != null)

```

Comment:
We are checking if `operation.Return?.Type` is null and then return `operation.Return`. Isn't that a little odd? I'd have expected us to check if `operation.Return` is null instead...